test/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM iron/go:dev as build-env
+
+ADD . /src
+
+RUN go get -d github.com/ethresearch/sharding-p2p-poc/...  
+    go get ./... &&\
+    make deps &&\
+    go build 
+
+COPY .sharding-p2p-poc    .
+
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o sharding-p2p-poc .
+
+FROM iron/go:dev
+
+RUN apk --no-cache add ca-certificates
+
+WORKDIR /root/
+
+COPY --from=0 /go/src/github.com/ethresearch/sharding-p2p-poc/    .
```

Comment:
It looks like 1-stage build is already awesome in this use case.
The main purpose to use 2-stage build is to put source code in one fat image for a disposable build environment and the binary built in another thin image for execution environment.